File: typescipt/src/specifications/iSpecificationRunnerContext.ts


export interface ISpecificationRunnerContext {
   number Failed
   IReadOnlyCollection<ISpecificationFileRunner> FileRunners

   void Fail(Scenario scenario, string message);
   void LogGlobal(string message);
   void Log(string message);
   void Success(Scenario scenario);
   void Add(ISpecificationFileRunner fileRunner);
   Array<IScenarioRunner> FailedScenariosRunners();
   number CountScenarios();
}

File: src/Lexy.Compiler/Specifications/ISpecificationRunnerContext.cs
using System.Collections.Generic;
using Lexy.Compiler.Language.Scenarios;
using Lexy.RunTime;

namespace Lexy.Compiler.Specifications;

public interface ISpecificationRunnerContext
{
    int Failed { get; }
    IReadOnlyCollection<ISpecificationFileRunner> FileRunners { get; }

    void Fail(Scenario scenario, string message, IEnumerable<string> errors);
    void LogGlobal(string message);
    void LogTimeSpent();

    void Success(Scenario scenario, IEnumerable<ExecutionLogEntry> logging = null);
    void Add(ISpecificationFileRunner fileRunner);
    IEnumerable<IScenarioRunner> FailedScenariosRunners();
    int CountScenarios();
}